litestar/handlers/websocket_handlers/listener.py

```diff
@@ -146,10 +201,33 @@ def __init__(
         self.dto = dto
         self.return_dto = return_dto
 
+        if not self.dependencies:
+            self.dependencies = {}
+        self.dependencies = dict(self.dependencies)
+        self.dependencies["lifespan_stub__"] = _utils.create_stub_dependency(
+            self._connection_lifespan or self.default_connection_lifespan
+        )
+        if self.on_accept:
+            self.dependencies["on_accept_dependencies"] = _utils.create_stub_dependency(self.on_accept.ref.value)
+        if self.on_disconnect:
+            self.dependencies["on_disconnect_dependencies"] = _utils.create_stub_dependency(
+                self.on_disconnect.ref.value
+            )
+
```

Comment:
I am aware that this is a bit hacky, but it's the cleanest solution overall I think